src/ser/mod.rs

```diff
@@ -163,7 +167,13 @@ impl Serializer {
         if let Some((ref config, ref mut pretty)) = self.pretty {
             pretty.indent += 1;
             if pretty.indent < config.depth_limit {
-                self.output += &config.new_line;
+                let is_empty = match self.is_empty {
```

Comment:
could be `unwrap_or(false)`

Reply:
Thanks, I'm still learning all the ways of Rust, `unwrap_or(false)` will make that a lot simpler.